src/main/java/com/dev/cinema/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,44 @@
+package com.dev.cinema.controller;
+
+import com.dev.cinema.dto.ShoppingCartResponseDto;
+import com.dev.cinema.mapper.ShoppingCartMapper;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.MovieSessionService;
+import com.dev.cinema.service.ShoppingCartService;
+import com.dev.cinema.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final MovieSessionService movieSessionService;
+    private final UserService userService;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
+                                  MovieSessionService movieSessionService,
+                                  UserService userService, ShoppingCartMapper shoppingCartMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.movieSessionService = movieSessionService;
+        this.userService = userService;
+        this.shoppingCartMapper = shoppingCartMapper;
+    }
+
+    @PostMapping("/movie-sessions")
+    public void addMovieSession(@RequestParam Long movieSessionId,
+                                @RequestParam Long userId) {
+        shoppingCartService.addSession(movieSessionService.getById(movieSessionId),
+                userService.getById(userId));
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getByUserId(@RequestParam(name = "id") Long userId) {
```

Comment:
```suggestion
    public ShoppingCartResponseDto getByUserId(@RequestParam Long userId) {
```
let's leave name userId(or you can change it to user_id if you want). just "id" would means that you are getting shopping cart by its own ID

Reply:
done